% -*- mode: noweb; noweb-default-code-mode: R-mode; -*-
%\VignetteIndexEntry{scHSQ}
%\VignetteKeywords{scRNA-Seq, evolution, celltype}
%\VignetteDepends{Biobase,monocle3,ggplot2}
%\VignettePackage{scHSQ}
%documentclass[12pt, a4paper]{article}
\documentclass[12pt]{article}

\textwidth=6.2in
\textheight=8.5in
%\parskip=.3cm
\oddsidemargin=.1in
\evensidemargin=.1in
\headheight=-.3in

\newcommand{\scscst}{\scriptscriptstyle}
\newcommand{\scst}{\scriptstyle}
\newcommand{\Rfunction}[1]{{\texttt{#1}}}
\newcommand{\Robject}[1]{{\texttt{#1}}}
\newcommand{\Rpackage}[1]{{\textit{#1}}}
\newcommand{\Rfunarg}[1]{{\textit{#1}}}

\author{UnJin Lee}
\begin{document}
\title{Description of the scHSQ package}

\maketitle
\tableofcontents
\section{Introduction}
\Rpackage{scHSQ} is a package designed to implement identification of a gene set that passes the HSQ criteria as described in Lee, et al (2025). 

A common issue of single cell transcriptomic analysis is in cell type identification across evolutionarily divergent species. Often, the transcriptomic divergence causes a strong species-specific effect that masks cell type effects such that cell types cannot be assigned across species. This is often true even when batch correction for species effect is applied. 

Lee, et al (2025) describes a simple methodology for identifying genes that possess high cell type information while simultaneously being evolutionarily conserved. UMAP projections on these genes that pass the HSQ criteria, as opposed to the entire transcriptome, allow for accurate cell type assignments across multiple species. This is accomplished by performing an analysis of variance (ANOVA) for known cell type and species labels on a gene-wise basis.

While the identification of genes passing the HSQ criteria is relatively simple, this package simplifies process by utilizing \Rpackage{monocle3}. Additionally, it allows for the visualization of gene expression on a single cross-species UMAP projection, a task that may be difficult without use of this \Rpackage{scHSQ} package. All that is needed run this analysis is a \Rpackage{monocle3} data set, known cell type labels for a reference species, and labels for species-of-origin for all other cells.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Getting started}
\label{sec:getstart}

To install the package: \\[6pt]
\texttt{R CMD INSTALL scHSQ\_x.y.z.tar.gz}\\[6pt]
\Rpackage{scHSQ} imports several functions from other packages. Make sure to have the following installed:\\
\Rpackage{Biobase}, \Rpackage{monocle3} and \Rpackage{ggplot2}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{First Steps}
<<echo=false,results=hide>>=
library(Biobase)
library(monocle3)
library(ggplot2)
@

For demonstration purposes we use a test data set provided by this package. To do this, we must first import the library, then import the data set. It is important that the transcriptomes for each species be aligned to each other. In Lee, et al (2025), and thus the data set provided below, this is performed by retaining only one-to-one reciprocal best orthologs. This data set was produced using orthology information provided by FlyBase. Alternatively, other orthology-calling software may be used to perform this step, including the quick and accurate DIAMOND algorithm, e.g. Buchfink, Reuter, and Drost (2021). 
<<results=hide>>=
library(scHSQ)
data(testis_3sp)
@

This data contains single-cell RNA-Seq data from three species: \textit{D. melanogaster}, \textit{D. yakuba}, and \textit{D. ananassae}. The data is wrapped in a \Rpackage{monocle3} package with species information encoded in the \texttt{spec} field and cell type information encoded in the \texttt{known\_type} field, with reference species labeled as \texttt{mel}. Each species' data set was individually imported using \Rpackage{monocle3}'s \texttt{new\_cell\_data\_set} function, reduced to retain only one-to-one orthologs, then combined using \Rpackage{monocle3}'s \texttt{combine\_cds} function. Subsequently, \Rpackage{monocle3}'s \texttt{preprocess\_cds}, \texttt{reduce\_dimension}, and \texttt{cluster\_cells} functions were applied. 

The resulting object is called \texttt{testis\_3sp}. We thus use the known labels to generate a new \Rpackage{scHSQ} object as following:

<<>>=
dros_testis <- new_scHSQ(testis_3sp, "mel", "spec", "known_type")
@


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Applying HSQ Criteria}

We now utilize the new \Rpackage{scHSQ} object to generate and apply our marker gene set. In this step, we calculate the ANOVA F-statistic for species and cell type on a gene-wise basis. We utilize median thresholds by default, but this may be altered either when initializing the new scHSQ object or through the use of \texttt{setPThresh}. This step may take a while, depending on the size of the input data sets.

<<results=hide>>=
dros_testis <- genMarkGenes(dros_testis)
dros_testis <- applyMarkGenes(dros_testis)
@

The cell type-specific divergence of each species' transcriptomes may be analyzed by plotting the \textit{log(F-statistic)} values for each gene. A high correlation suggests a strong degree of cell type-specific evolution. 

<<results=hide>>=
plotANOVA(dros_testis)
@ 

\begin{figure}[h!]
\centering
\includegraphics[width=0.5\textwidth]{scHSQ_ANOVA}
\caption{Strong cell type-specific evolution in \textit{Drosophila} testis tissue.}
\end{figure}

After these commands complete, a cross-species UMAP projection should be stored in the \texttt{reducedCDS} slot of the \Rpackage{scHSQ} object. We can extract the list of marker genes utilized for this UMAP projection. Additionally, we can produce plots on the cross-species UMAP projection. In this case, the \texttt{ctplots} object is a list of UMAP projections containing: 1) known cell type labels for the reference species, 2) new clusters for all cells across all species, and 3) clusters for cells in each respective species. Note that arguments for \Rpackage{monocle3}'s \texttt{plot\_cells} can be passed to \Rpackage{scHSQ}'s \texttt{plotCTAssign} function and will automatically be applied to all elements of \texttt{ctplots}.  

<<results=hide>>=
new_CDS <- getReducedCDS(dros_testis)
gene_list <- getMarkGenes(dros_testis)

ctplots <- plotCTAssign(dros_testis)
@

\begin{figure}[h!]
\centering
\includegraphics[width=0.5\textwidth]{scHSQ_UMAP_ref}
\caption{Single UMAP of original cell type assignments for reference species. This is found in \texttt{ctplots[[1]]}.}
\end{figure}

\begin{figure}[h!]
\centering
\includegraphics[width=0.5\textwidth]{scHSQ_UMAP_all}
\caption{Single UMAP of new cell type assignments for all species. This is found in \texttt{ctplots[[2]]}.}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Cell type assignments across species}

The UMAP figures from the previous section suggest a clear mapping of reference cell type assignments to the newly computed cell type assignments. If these mappings prove to be unsatisfactory, you may wish to adjust the HSQ thresholds using the \texttt{setPThresh} and/or recluster on the new UMAP with a different resolution using the \texttt{reclusterCDS} function. Once a satisfactory assignment has been achieved, the cell type assignments should be remapped using \texttt{remapCTAssign}. We may then plot the resulting re-assigned clusters.

<<results=hide>>=
old_clu <- as.character(1:10)
new_clu <- c("Late spermatocytes", "Early spermatocytes", "Early spermatids", 
"Early spermatocytes", "Early spermatids", "Late spermatogonia", "Somatic", 
"Late spermatids", "ananassae spermatocytes", "GSC Early spermatogonia")
cell_types <- data.frame(HSQ_clu=old_clu, HSQ_clu1=new_clu)

dros_testis <- remapCTAssign(dros_testis, cell_types)

ctplots_reassign <- plotCTAssign(dros_testis, "HSQ_clu1")	
@

\begin{figure}[h!]
\centering
\includegraphics[width=0.5\textwidth]{scHSQ_UMAP_reassign}
\caption{Single UMAP of re-assigned cell types for reference species. This is found in \texttt{ctplots\_reassign[[2]]}.}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Gene expression on new UMAP}

Finally, we may be interested in seeing a single gene's expression on the new UMAP. To generate this figure, we use the example of \textit{Rbp4}, a gene that did not pass the HSQ criteria. We utilize the \texttt{plotGeneExprs} function to generate this figure. As before, we are also able to forward arguments for \Rpackage{monocle3}'s \texttt{plot\_cells} function. We may also generate this plot for a single species' cells by using the \texttt{species=} argument.


<<results=hide>>=
gene_exprs <- plotGeneExprs(dros_testis, "Rbp4", alpha=0.5, 
norm_method="size_only", min_expr=40)
@

\begin{figure}[h!]
\centering
\includegraphics[width=0.5\textwidth]{scHSQ_UMAP_Rbp4}
\caption{Expression of \textit{Rbp4} on single UMAP of across species.}
\end{figure}


\newpage
\appendix
REFERENCES

Lee U, Li C, Langer CB, Svetec N, Zhao L (2025) Comparative Single Cell Analysis of Transcriptional Bursting Reveals the Role of Genome Organization on de novo Transcript Origination. bioRxiv doi:10.1101/2024.04.29.591771

Buchfink B, Reuter K, Drost H-G (2021), Sensitive protein alignments at tree-of-life scale using DIAMOND. Nature Methods, 18: 366-368 doi:10.1038/s41592-021-01101-x

\end{document}












